src/algorithms/piecewise_linear_stretching.jl

```diff
@@ -0,0 +1,179 @@
+"""
+```
+    PiecewiseLinearStretching <: AbstractHistogramAdjustmentAlgorithm
+    PiecewiseLinearStretching(;  src_intervals = (0.0 => 0.1, 0.1 => 0.9, 0.9 => 1.0), 
+                                 dst_intervals = (0.0 => 0.2, 0.2 => 1.0, 1.0 => 1.0))

```

Comment:
I'm not sure how I feel about the current Pair specification. My main confusion is that `0.0` doesn't map to `0.1`; instead, it is a bounded tuple that specifies the start and end value of the interval. If we want to express the "map" meaning, we might use `src_range_1 => dst_range_1, src_range_2 => dst_range_2`

How about

```julia
PiecewiseLinearStretching((0.0, 0.1) => (0.0, 0.2), (0.1, 0.9)=>(0.2, 1.0), (0.9, 1.0)=>(1.0, 1.0))
```

?

This naturally degenerates to `LinearStretching` if we only pass one pair of tuple:

```julia
# these are equivalent
PiecewiseLinearStretching((0.0, 1.0)=>(0.0, 1.0))
LinearStretching((0.0, 1.0)=>(0.0, 1.0))
```

---

This might be a good chance to introduce IntervalSets to simplify the specification, e.g.,:

```julia
PiecewiseLinearStretching(0.0..0.1 => 0.0..0.2, 0.1..0.9=>0.2..1.0, 0.9..1.0=>1.0..1.0)
```

which would possibly make argument checking easier.